woocommerce/woocommerce
# Data: add selector to get related products

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR added the `getRelatedProduct()` selector to the `wc/admin/products` store. 

Needed in https://github.com/woocommerce/woocommerce/issues/42922
Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure you have a few products to test
2. Enabled the new Product Editor
3. Identify the ID of a product you have.
4. Go to the edit page of the new product editor
5. Open the dev console
6. Knowing the product ID, select the related products by running the following command:
```js
wp.data.select( 'wc/admin/products' ).getRelatedProducts( <product-ID )
```
In my case, I know the product ID is `17`, so...

```js
wp.data.select( 'wc/admin/products' ).getRelatedProducts( 17 )
```

7. Confirm you see the client performing an async request to get the products list

payload | response
--------|------
<img width="445" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/dfd4be0a-3a8c-49b7-ad67-eb5dafb0c151"> | <img width="600" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/ab40d612-3cd4-4c5d-8ad8-3808b394d6bb">

8. re-run the same command to see the related products in the console:
```js
wp.data.select( 'wc/admin/products' ).getRelatedProducts( 17 )
```
9. Confirm the client does not perform an addition request
<img width="519" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/04159e4f-3776-4821-8fef-b1b9fb8cc3ae">




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
